sql/time_bucket.sql

```diff
@@ -30,39 +39,21 @@ $BODY$
     SELECT (@extschema@.time_bucket(bucket_width, ts-"offset")+"offset")::date;
 $BODY$;
 
-
-CREATE OR REPLACE FUNCTION time_bucket(bucket_width BIGINT, ts BIGINT)
-    RETURNS BIGINT LANGUAGE SQL IMMUTABLE PARALLEL SAFE AS
+CREATE OR REPLACE FUNCTION time_bucket(bucket_width SMALLINT, ts SMALLINT, "offset" SMALLINT)
+    RETURNS SMALLINT LANGUAGE SQL IMMUTABLE PARALLEL SAFE AS
 $BODY$
-    SELECT (ts / bucket_width)*bucket_width;
+    SELECT @extschema@.time_bucket(bucket_width, ts-"offset")+"offset";
```

Comment:
Shouldn't we check for overflow here? For instance, raise error of overflowing. Or does PG somehow magically handle that (I guess it could since its internal sub functions check for that)? Maybe add tests?

Reply:
PG catches this.

Reply:
i've added tests for this